compiler/modules/front.syntax/token.bal

```diff
@@ -249,6 +251,16 @@ class Tokenizer {
         return createPosition(self.lineIndex, self.tokenStartCodePointIndex);
     }
 
+    function previousPos() returns Position {
```

Comment:
I don't like this at all.  Too stateful. Instead we should have a function that returns the Position of the end of the current token (currentPos() returns the start of the current token).  We should probably rename currentPos() to currentStartPos(); then the new function would be currentEndPos(). An endPos should always come from calling currentEndPos() and a startPos should always come from calling currentStartPos() (otherwise whitespace will be in the wrong place).

Reply:
fixed